x-pack/plugins/cloud_security_posture/public/common/navigation/use_csp_breadcrumbs.ts

```diff
@@ -27,47 +23,53 @@ const getClickableBreadcrumb = (
   }
 };
 
-export const useCspBreadcrumbs = (breadcrumbs: CspNavigationItem[]) => {
+export const useCspBreadcrumbs = (breadcrumbs: BreadcrumbEntry[]) => {
   const {
     services: {
       chrome: { setBreadcrumbs, docTitle },
-      application: { getUrlForApp },
+      application: { currentAppId$, applications$, navigateToApp },
     },
   } = useKibana<CoreStart>();
+
   const match = useRouteMatch();
-  const history = useHistory();
+
+  const appId = useObservable(currentAppId$);
+  const applications = useObservable(applications$);
+  const application = appId ? applications?.get(appId) : undefined;
+  const appTitle = application?.title;
 
   useEffect(() => {
     const additionalBreadCrumbs: ChromeBreadcrumb[] = breadcrumbs.map((breadcrumb) => {
       const clickableLink = getClickableBreadcrumb(match.path, breadcrumb.path);
 
       return {
         text: breadcrumb.name,
-        ...(clickableLink && {
-          onClick: (e) => {
-            e.preventDefault();
-            history.push(clickableLink);
-          },
-        }),
+        ...(clickableLink &&
+          appId && {
+            onClick: (e) => {
+              e.preventDefault();
+              void navigateToApp(appId, { path: clickableLink });
```

Comment:
what is the `void` doing? just curious

Reply:
`void` evaluates an expression and returns `undefined` - I use it when I want to ignore the result of a promise to make sure the intent to not use the promise return value is clear - inspired by [this ESLint rule](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-floating-promises.md#ignorevoid)